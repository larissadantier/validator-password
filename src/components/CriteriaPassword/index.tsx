import List from './List';
import Item from './Item';
import Title from './Title';
import Text from './Text';

export const CriteriaPassword = {
  List,
  Item,
  Title,
  Text
}