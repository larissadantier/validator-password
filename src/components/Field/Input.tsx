import { ComponentProps } from "react";

import styles from './field.module.scss';

function Input({ ...props }: ComponentProps<'input'>) {
  return (
    <input data-testid="field__input" className={styles['field__input']} autoComplete='off' {...props} />
  );
}

export default Input;