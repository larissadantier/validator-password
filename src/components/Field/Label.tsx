import { ComponentProps } from "react";

function Label({ children, ...props }: ComponentProps<'label'>) {
  return (
    <label data-testid="field__label" {...props}>{children}</label>
  );
}

export default Label;