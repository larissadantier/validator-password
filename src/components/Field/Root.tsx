import { ComponentProps } from "react";

import styles from './field.module.scss';

function Root({ children, ...props }: ComponentProps<'div'>) {
  return (
    <div data-testid="field" className={styles['field__container']} {...props}>
      {children}
    </div>
  );
}

export default Root;