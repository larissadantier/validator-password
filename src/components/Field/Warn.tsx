import { ComponentProps } from "react";

import styles from './field.module.scss';

function Warn({ children }: ComponentProps<'span'>) {
  return (
    <span data-testid="field__warn" className={styles['field__warn']}>{children}</span>
  );
}

export default Warn;