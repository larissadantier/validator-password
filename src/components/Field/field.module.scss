.field {
  &__container {
    display: flex;
    flex-direction: column;
    gap: 0.5rem;

    font-size: 1rem;
  }

  &__warn {
    display: block;
    color: var(--error);
    font-size: 14px;
  }

  &__input {
    border: 1px solid var(--gray-primary);
    border-radius: 5px;

    padding: 0.5rem;

    box-shadow: var(--box-shadow) 0px 2px 8px 0px;

    &:focus-within {
      border-color: var(--focus-blue);
      box-shadow: var(--focus-blue-shadow) 0px 2px 8px 0px;
    }

    &::placeholder {
      color: var(--gray-secondary);
    }
  }
}