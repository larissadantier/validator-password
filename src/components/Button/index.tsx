import Root from "./Root";
import Text from './Text';

export const Button = {
  Root,
  Text
}