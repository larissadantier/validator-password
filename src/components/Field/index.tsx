import Root from './Root';
import Label from './Label';
import Input from "./Input";
import Warn from './Warn';


export const Field = {
  Root,
  Label,
  Input,
  Warn
}