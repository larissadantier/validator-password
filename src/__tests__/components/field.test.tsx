import { expect, describe, it } from 'vitest';
import { fireEvent, render } from '@testing-library/react';

import { Field } from '../../components/Field';

const fieldTestId = 'field';
const fieldInputTestId = 'field__input';
const fieldLabelTestId = 'field__label';
const fieldWarnTestId = 'field__warn';

describe('Component Field', () => {
  it('should be able render the field', () => {
    const { getByTestId } = render(
      <Field.Root>
        <Field.Label>Nome</Field.Label>
        <Field.Input />
      </Field.Root>
    );

    expect(getByTestId(fieldTestId)).toBeInTheDocument();
  })

  it('should be able to see a label', () => {
    const { getByTestId } = render(
      <Field.Root>
        <Field.Label>Nome</Field.Label>
        <Field.Input />
      </Field.Root>
    );

    const label = getByTestId(fieldLabelTestId)

    expect(label).toBeInTheDocument();

    expect(label).toHaveTextContent('Nome');
  })

  it('should be able to see a placeholder', () => {
    const { getByPlaceholderText } = render(
      <Field.Root>
        <Field.Label>Nome</Field.Label>
        <Field.Input placeholder='Escreva o seu nome' />
      </Field.Root>
    );

    const input = getByPlaceholderText('Escreva o seu nome');

    expect(input).toBeInTheDocument();
  })

  it('should be able update the value when typing in the input', () => {
    const { getByTestId } = render(
      <Field.Root>
        <Field.Label>Nome</Field.Label>
        <Field.Input placeholder='Escreva o seu nome' />
      </Field.Root>
    );

    const input = getByTestId(fieldInputTestId) as HTMLInputElement;

    fireEvent.change(input, { target: { value: 'Larissa Dantier' } });

    expect(input.value).toBe('Larissa Dantier');
  })

  it('should be able to see a focus using table or click', () => {
    const { getByTestId } = render(
      <Field.Root>
        <Field.Label>Nome</Field.Label>
        <Field.Input placeholder='Escreva o seu nome' />
      </Field.Root>
    )

    const input = getByTestId(fieldInputTestId)

    input.focus()

    expect(input).toHaveFocus()
  })

  it('should be able to see a warn for error input', () => {
    const { getByTestId } = render(
      <Field.Root>
        <Field.Label>Nome</Field.Label>
        <Field.Input placeholder='Escreva o seu nome' />

        <Field.Warn>Preencha o campo de nome</Field.Warn>
      </Field.Root>
    )

    const warn = getByTestId(fieldWarnTestId);

    expect(warn).toBeInTheDocument();
    expect(warn).toHaveTextContent('Preencha o campo de nome');
  })

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Field.Root>
        <Field.Label>Nome</Field.Label>
        <Field.Input placeholder='Escreva o seu nome' />

        <Field.Warn>Preencha o campo de nome</Field.Warn>
      </Field.Root>
    );

    expect(asFragment()).toMatchSnapshot();
  });
})