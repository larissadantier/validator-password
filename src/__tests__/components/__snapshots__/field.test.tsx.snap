// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component Field > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="_field__container_cac991"
    data-testid="field"
  >
    <label
      data-testid="field__label"
    >
      Nome
    </label>
    <input
      autocomplete="off"
      class="_field__input_cac991"
      data-testid="field__input"
      placeholder="Escreva o seu nome"
    />
    <span
      class="_field__warn_cac991"
      data-testid="field__warn"
    >
      Preencha o campo de nome
    </span>
  </div>
</DocumentFragment>
`;
